lib/hypothesis.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'hypothesis/errors'
+require 'hypothesis/providers'
+require 'hypothesis/engine'
+
+module Hypothesis
+  def hypothesis(options = {}, &block)
+    unless @current_hypothesis_engine.nil?
+      raise UsageError, 'Cannot nest hypothesis calls'
+    end
+    begin
+      @current_hypothesis_engine = Engine.new(options)
```

Comment:
one thing to be aware of: this means that any object that has this module mixed in will have `@current_hypothesis_engine` available to it as an instance variable. If you don't want users to be able to do this, it might be worth defining a `Hypothesis::World` module and storing state there, which wont' affect user objects.

Reply:
(this only applies to modules, and not classes)